src/main/drivers/bus_i2c_hal.c

```diff
@@ -182,11 +187,23 @@ bool i2cRead(I2CDevice device, uint8_t addr_, uint8_t reg_, uint8_t len, uint8_t
 
 void i2cInit(I2CDevice device)
 {
-    /*## Configure the I2C clock source. The clock is derived from the SYSCLK #*/
-//    RCC_PeriphCLKInitTypeDef  RCC_PeriphCLKInitStruct;
-//    RCC_PeriphCLKInitStruct.PeriphClockSelection = i2cHardwareMap[device].clk;
-//    RCC_PeriphCLKInitStruct.I2c1ClockSelection = i2cHardwareMap[device].clk_src;
-//    HAL_RCCEx_PeriphCLKConfig(&RCC_PeriphCLKInitStruct);
+    if (device == I2CINVALID) {
+        return;
+    }
+
+    i2cDevice_t *pDev = &i2cDevice[device];
+
+    const i2cHardware_t *hardware = pDev->hardware;
+
+    if (!hardware) {
+        return;
+    }
+
+    IO_t scl = IOGetByTag(pDev->scl);
+    IO_t sda = IOGetByTag(pDev->sda);
+
+    IOInit(scl, OWNER_I2C_SCL, RESOURCE_INDEX(device));
+    IOInit(sda, OWNER_I2C_SDA, RESOURCE_INDEX(device));
 
     switch (device) {
     case I2CDEV_1:
```

Comment:
One of them is not necessary (but it does not harm). Personally I'll kill the `switch` variant